drivers/rtc/rtc_max31329.c

```diff
@@ -0,0 +1,570 @@
+		LOG_ERR(" ID %d not supporting", id);
+		return -ENOTSUP;
+	}
+
+	/* id =0 mapped to Alarm 1*/
+	/* id =1 mapped to Alarm 2*/
+	const uint16_t valid_mask =
+		id ? MAX31329_RTC_ALARM_2_TIME_MASK : MAX31329_RTC_ALARM_1_TIME_MASK;
+
+	if ((mask & ~valid_mask) != 0) {
+		LOG_ERR("invalid alarm field mask 0x%04x", mask);
+		return -EINVAL;
+	}
+
+	/*
+	 * The 7/6 bit is used as enabled/disabled flag.
+	 * The mask will clean it and also the unused bits
+	 */
+	/* Alarm 1 settings*/
+	/*  SEC*/
+	if ((mask & RTC_ALARM_TIME_MASK_SECOND) != 0) {
+		regs[0] = bin2bcd(timeptr->tm_sec) & ~BIT(7);
+	} else {
+		LOG_DBG("SEC FLAG NOT SET");
+		regs[0] = BIT(7);
+	}
+
+	/*  MIN */
+	if ((mask & RTC_ALARM_TIME_MASK_MINUTE) != 0) {
+		regs[1] = bin2bcd(timeptr->tm_min) & ~BIT(7);
+	} else {
+		LOG_DBG("MIN FLAG NOT SET");
+		regs[1] = BIT(7);
+	}
+
+	/*  HR*/
+	if ((mask & RTC_ALARM_TIME_MASK_HOUR) != 0) {
+		regs[2] = bin2bcd(timeptr->tm_hour) & (MAX31329_HOURS_MASK & ~BIT(7));
+	} else {
+		LOG_DBG("HR FLAG NOT SET");
+		regs[2] = BIT(7);
+	}
+
+	/*  MONTH DAY*/
+	if ((mask & RTC_ALARM_TIME_MASK_MONTHDAY) != 0) {
+		regs[3] = bin2bcd(timeptr->tm_mday) & (MAX31329_DAYS_MASK & ~(BIT(7) | BIT(6)));
+	} else {
+		LOG_DBG("MONTH DAY FLAG NOT SET");
+		regs[3] = BIT(7);
+	}
+
+	/*  MONTH*/
+	if ((mask & RTC_ALARM_TIME_MASK_MONTH) != 0) {
+		regs[4] = bin2bcd((timeptr->tm_mon) + 1) & (MAX31329_MONTHS_MASK & ~BIT(7));
+	} else {
+		LOG_DBG("MONTH FLAG NOT SET");
+		regs[4] = BIT(7);
+	}
+
+	/*  YEAR*/
+	if ((mask & RTC_ALARM_TIME_MASK_YEAR) != 0) {
+		regs[5] = bin2bcd(timeptr->tm_year) & ~BIT(7);
+	} else {
+		LOG_DBG("YEAR FLAG NOT SET");
+		regs[4] |= BIT(6);
+		regs[5] = 0x00;
+	}
+
+	if (!id) {
+		ret = i2c_burst_write_dt(&config->i2c, MAX31329_REG_ALM1_SEC, regs, sizeof(regs));
+		if (ret) {
+			LOG_ERR("Error when setting alarm 1: %i", ret);
+			return ret;
+		}
+
+	} else {
+		ret = i2c_burst_write_dt(&config->i2c, MAX31329_REG_ALM2_MIN, &regs[1], 3);
+		if (ret) {
+			LOG_ERR("Error when setting alarm 2: %i", ret);
+			return ret;
+		}
+	}
+
+	/* Enable RTC side interrupts,*/
+	i2c_reg_read_byte_dt(&config->i2c, MAX31329_REG_INT_EN, &regs[0]);
+	regs[0] |= BIT(id ? 1 : 0);
+
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_INT_EN, regs[0]);
+	if (ret) {
+		LOG_ERR("Error when enabling interrupts: %i", ret);
+		return ret;
+	}
+
+	/* set intB as Alarm interrupts: Note Table 2 in data sheet*/
+	regs[0] = 0x00;
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_RTC_CONFIG2, regs[0]);
+	if (ret) {
+		LOG_ERR("Error when setting intB: %i", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int max31329_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *mask,
+				   struct rtc_time *timeptr)
+{
+	const struct max31329_config *config = dev->config;
+	int ret = 0;
+	uint8_t raw_time[7] = {0};
+
+	if (id != 0) {
+		LOG_ERR(" ID %d not supporting", id);
+		return -ENOTSUP;
+	}
+
+	/*starting from seconds register read till year*/
```

Comment:
Inconsistent comments, Seems most use Capitalization and space before and after text
`/* Example */`, please use the same style everywhere :)